### ISSUE TITLE
Inconsistent keyboard behaviour in post editor and widgets editor headers

### ISSUE THREAD

**Issue by tellthemachines** [WordPress/gutenberg#38257]
### Description

In both the post and the widgets editor, tabbing into the editor from the "skip to content" link lands us first on the block inserter toggle, and from there the next tab goes straight into the header settings section ("Switch to draft", "Preview" and so on). In order to access the "Document tools" section, we have to use the arrow keys after landing on the block inserter toggle.

The problem is that there's no hint that the arrow keys are necessary to navigate this section, and in the site editor the same section is navigable with the Tab key, so behaviour is also inconsistent between editors.

It should be possible to navigate the whole header with just the Tab key in all the editors.

### Step-by-step reproduction instructions

1. Open a post in the block editor
2. Tab once to land on "Skip to main content" button, then press Enter and Tab again to land on block inserter toggle
3. Try pressing Tab again and verify that focus goes straight to header settings area.

### Please confirm that you have searched existing issues in the repo.

Yes

### Please confirm that you have tested with all plugins deactivated except Gutenberg.

Yes
